### PR TITLE
A mod that let you edit webhook

### PR DESCRIPTION
**Pull request by marceaupatu123**
**You can edit webhook so it could be usefull, i did it after a request  (https://discord.com/channels/374961173524643843/374961342328602637/751886830554841125).**

**Status**

- [x] Code changes have been tested against the Discord API and the discord.js wrapper, or there are no code changes
- [x] Documentation has been added/modified, or there is nothing to change (docs/mods.json)

**Semantic versioning classification:**

- [ ] This PR changes DBM's interface (methods or parameters added to default methods)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Review by almeidx — CHANGES_REQUESTED** [review]
Follow the [modification template](https://github.com/dbm-network/documents/blob/master/template_MOD.js).

**Comment by master3395** [comment]
There seems to be an issue with this version.

image

**Comment by marceaupatu123** [comment]
> There seems to be an issue with this version.
> Every time i try to start the bot,
> I get this message.
> If i remove this action, it starts fine.
> 
> image

Let me convert ES6 code to commonjs

**Comment by marceaupatu123** [comment]
@almeidx @master3395 Fixed

**Review by marceaupatu123 — COMMENTED** [review]
Changed for @almeidx and @master3395

File: actions/edit_webhook.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    const wh = Mods.getWebhook(webhook, varName, cache)
+    if (wh) {
+      const avatar = this.evalMessage(data.webhookIcon, cache)
+      const name = this.evalMessage(data.webhookName, cache)
+      if (avatar) {
+        wh.edit({ avatar })
+      }
+      if (name) {
+        wh.edit({ name })
+      }
```
Comment (almeidx) [inline actions/edit_webhook.js:54]: If they provide an avatar and a name it will make 2 api calls. It should only make one at all times

**Review by almeidx — CHANGES_REQUESTED** [review]

**Review by marceaupatu123 — COMMENTED** [review]
Now, only 1 api call.

File: actions/edit_webhook.js (hunk 1 above)
Comment (marceaupatu123) [reply actions/edit_webhook.js]: Okay, now it's fixed.

File: actions/edit_webhook.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      if (avatar && name) {
+        wh.edit({ avatar, name })
+      } else if (avatar) {
+        wh.edit({ avatar })
+      } else if (name) {
+        wh.edit({ name })
+      }
+      this.callNextAction(cache)
+    } else {
+      this.callNextAction(cache)
```
Comment (almeidx) [inline actions/edit_webhook.js:58]: This else is unnecessary. Simply call the callNextAction method after the if (and remove it inside the if as well)

**Review by almeidx — CHANGES_REQUESTED** [review]

File: actions/edit_webhook.js (hunk 2 above)
Comment (marceaupatu123) [reply actions/edit_webhook.js]: Is it better now ?

**Review by almeidx — APPROVED** [review]

**Pull request merged**